awx/main/models/events.py

```diff
@@ -546,6 +546,17 @@ def _update_host_summary_from_stats(self, hostnames):
 
             Host.objects.bulk_update(list(updated_hosts), ['last_job_id', 'last_job_host_summary_id'], batch_size=100)
 
+            updated_hosts_list = list()
+            for host in updated_hosts:
+                updated_hosts_list.append(host.name)
+
+            # bulk-create
+            current_time = now()
+            HostMetrics.objects.bulk_create(
+                [HostMetrics(hostname=hostname, last_automation=current_time) for hostname in updated_hosts_list], ignore_conflicts=True
```

Comment:
We might also want to take advantage of the Django ORM's support for batching these inserts *and* updates:

https://docs.djangoproject.com/en/3.1/ref/models/querysets/#django.db.models.query.QuerySet.bulk_create

`batch_size=100`, maybe?

In other places, this is what we've done:

https://github.com/ansible/awx/blob/devel/awx/main/models/events.py#L547

Reply:
@ryanpetrello not sure if we're using `django-postgres-extra` anywhere, but with that we can do 1 query that will insert or update on conflict like: https://django-postgres-extra.readthedocs.io/en/master/conflict_handling.html#bulk

Reply:
😉 

https://github.com/ansible/tower/issues/4687#issuecomment-807375828

We decided to just take this approach instead of introducing a new dependency.

Reply:
@Ladas we are not.  `django-postgres-extra` came up in another context recently, but we decided against bringing it in as a new dependency.

Reply:
I have made the required changes